tests/loaders/osm_tag_loader/test_osm_tag_loader.py

```diff
@@ -0,0 +1,208 @@
+        elif polygon == empty_polygon:
+            return gpd.GeoDataFrame(crs=WGS84_CRS, geometry=[])
+        return None
+
+    mocker.patch("osmnx.geometries_from_polygon", new=mock_geometries_from_polygon)
+
+
+@pytest.fixture  # type: ignore
+def expected_result_single_polygon() -> gpd.GeoDataFrame:
+    """Get the expected result of a query with single polygon for testing."""
+    return gpd.GeoDataFrame(
+        {
+            "amenity": ["restaurant"],
+        },
+        geometry=[Point(0, 0)],
+        index=pd.MultiIndex.from_arrays(
+            arrays=[
+                ["node"],
+                [1],
+            ],
+            names=["element_type", "osmid"],
+        ),
+        crs=WGS84_CRS,
+    )
+
+
+@pytest.fixture  # type: ignore
+def expected_result_gdf_simple() -> gpd.GeoDataFrame:
+    """Get the expected result of a simple query for testing."""
+    return gpd.GeoDataFrame(
+        {
+            "amenity": ["restaurant", "restaurant"],
+        },
+        geometry=[Point(0, 0), Point(1, 1)],
+        index=pd.MultiIndex.from_arrays(
+            arrays=[
+                ["node", "node"],
+                [1, 2],
+            ],
+            names=["element_type", "osmid"],
+        ),
+        crs=WGS84_CRS,
+    )
+
+
+@pytest.fixture  # type: ignore
+def expected_result_gdf_complex() -> gpd.GeoDataFrame:
+    """Get the expected result of complex query for testing."""
+    return gpd.GeoDataFrame(
+        {
+            "amenity": ["restaurant", "restaurant", "bar"],
+            "building": ["commercial", None, "retail"],
+        },
+        geometry=[Point(0, 0), Point(1, 1), Polygon([(0, 0), (1, 0), (1, 1), (0, 1)])],
+        index=pd.MultiIndex.from_arrays(
+            arrays=[
+                ["node", "node", "way"],
+                [1, 2, 3],
+            ],
+            names=["element_type", "osmid"],
+        ),
+        crs=WGS84_CRS,
+    )
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "area_gdf_fixture,query,expected_result_gdf_fixture",
+    [
+        ("single_polygon_area_gdf", {"amenity": "restaurant"}, "expected_result_single_polygon"),
+        ("two_polygons_area_gdf", {"amenity": "restaurant"}, "expected_result_gdf_simple"),
+        (
+            "two_polygons_area_gdf",
+            {"amenity": ["restaurant", "bar"], "building": True},
+            "expected_result_gdf_complex",
+        ),
+        (
+            "empty_area_gdf",
+            {"amenity": ["restaurant", "bar"], "building": True},
+            "empty_result_gdf",
+        ),
+        (
+            "area_with_no_objects_gdf",
+            {"amenity": ["restaurant", "bar"], "building": True},
+            "empty_result_gdf",
+        ),
+    ],
+)
+def test_osm_tag_loader(
+    area_gdf_fixture: str,
+    query: Dict[str, Union[List[str], str, bool]],
+    expected_result_gdf_fixture: str,
+    request: Any,
+):
+    """Test `OSMTagLoader.load()`."""
+    _ = request.getfixturevalue("mock_osmnx")
```

Comment:
You can add autouse=True to this fixture definition and then there is no need to ask for it like that

Reply:
Good to know, thanks. I won't bother adding it here though.